### PR TITLE
EVG-6293 min hosts not preserved for rhel70-stitch distro

### PR DESCRIPTION
**Pull request by StuartButler**
Now I remember why I had that second sort in the aggregation for `IdleEphemeralGroupedByDistroID()`

### CODE REVIEW

File: units/crons.go [hunk 1]
```diff
@@ -329,9 +329,8 @@ func PopulateIdleHostJobs(env evergreen.Environment) amboy.QueueOperation {
 		}
 
 		for _, info := range distroHosts {
-			distroID := info.DistroID
 			totalRunningHosts := info.RunningHostsCount
-			minimumHosts := distrosMap[distroID].PlannerSettings.MinimumHosts
+			minimumHosts := distrosMap[info.DistroID].PlannerSettings.MinimumHosts
```
Comment (bsamek) [inline units/crons.go]: Is it possible to add a test that covers this bug?

Comment (StuartButler) [reply units/crons.go]: Working on it

**Comment by tychoish** [comment]
this seems good, less the test brian asked for and a commit message that reflects the bug/cause without the specific case that tripped it (e.g. "preserve minimum hosts for distros"

File: units/crons.go (hunk 1 above)
Comment (StuartButler) [reply units/crons.go]: Added a test for initial review.

**Pull request merged**